UI Banner doesn't show error when using Authentication Provider

Internal reference: SURE-5454
Reported in 2.6.8

The user is using Active Directory Authentication Provider and the UI banner doesn't show any error message for failed authentication.

For local Authentication, if there is a failure, the UI banner shows the following error: 
*"Invalid username or password. Please try again."*

![image-2022-10-21-13-03-28-515.png](https://zube.io/files/rancher/aada407d568901d03928e5ba722c2d31-image-2022-10-21-13-03-28-515.png)

For Auth provider, if the authentication fails, the UI just shows "Error" on the button and not a message. as shown in the screenshot below:

![image-2022-10-21-13-03-34-120.png](https://zube.io/files/rancher/caaaed216189f3468eabf96c09b2d7cc-image-2022-10-21-13-03-34-120.png)

Looking into rancher pods, we can see the authentication failure reason is logged, for example, the following log entry shows that failure was due to `Invalid Credentials, code 532`, which means the password expired:

`2022/10/18 15:01:29 [ERROR] API error response 401 for POST /v3-public/activeDirectoryProviders/activedirectory?action=login. Cause: LDAP Result Code 49 "Invalid Credentials": 80090308: LdapErr: DSID-0C090436, comment: AcceptSecurityContext error, data 532, v4f7
`

Furthermore, the "Login Failed Banner" feature also doesn't work with Authentication Providers. That feature was added as part of https://github.com/rancher/dashboard/issues/5401

Impact:

The user can not identify if the logging failure error is due to the auth provider or due to rancher internals failure

Repro steps:

- Add Authentication Provider to Rancher (For example, ActiveDirectory)
- Try to authenticate using incorrect credentials

Actual behavior:

UI Banner doesn't show an error when using an Authentication Provider

Expected behavior:

UI Banner should show an error when using an Authentication Provider

--- a/shell/components/auth/login/ldap.vue
+++ b/shell/components/auth/login/ldap.vue
@@ -34,6 +34,9 @@ export default {
         this.$router.replace('/');
       } catch (err) {
         this.err = err;
+
+        // emit error to parent so that it can displayed on the error Banner
+        this.$emit('error', err);
         buttonCb(false);
       }
     },

--- a/shell/pages/auth/login.vue
+++ b/shell/pages/auth/login.vue
@@ -225,6 +225,10 @@ export default {
       }
     },
 
+    handleProviderError(err) {
+      this.err = err;
+    },
+
     async loginLocal(buttonCb) {
       try {
         await this.$store.dispatch('auth/login', {
@@ -383,6 +387,7 @@ export default {
             :name="name"
             :open="!showLocal"
             @showInputs="showLocal = false"
+            @error="handleProviderError"
           />
         </div>
         <template v-if="hasLocal">

